delvingbitcoin_2_elasticsearch/delvingbitcoin_2_elasticsearch.py

```diff
@@ -0,0 +1,76 @@
+import os
+import json
+from dotenv import load_dotenv
+
+# from loguru import logger
+from elastic import create_index, document_add, document_exist, document_view
+from achieve import download_dumps
+
+dotenv_path = os.path.join(os.path.dirname(__file__),'..','.env')
+load_dotenv(dotenv_path)
+
+# Get environment variables for path and index name
+INDEX = os.getenv("INDEX")
+ARCHIVE = os.getenv("ARCHIVE") or "archive"
+SUB_ARCHIVE = os.getenv("SUB_ARCHIVE") or "posts"
+
+# Create Index if it doesn't exist
+if create_index(INDEX):
+    print(f"Index: {INDEX}, created successfully.")
+else:
+    print(f"Index: {INDEX}, already exist.")
+
+# Get the directory where the Python script is located
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+# Specify the path to the folder containing JSON files
+folder_path = os.path.join(script_dir, ARCHIVE, SUB_ARCHIVE)
+
+
+def index_documents(files_path):
+    # Iterate through files in the specified path
+    for root, dirs, files in os.walk(files_path):
+        for file in files:
+            if file.endswith('.json'):
+                file_path = os.path.join(root, file)
+                print(f'Fetching document from {file_path}')
+
+                # Load JSON data from file
+                with open(file_path, 'r', encoding='utf-8') as json_file:
+                    document = json.load(json_file)
+
+                # Select required fields
+                doc = {
+                    'id': f'{document["id"]}_{document["username"]}_{document["topic_slug"]}_{document["post_number"]}',
+                    'authors': document['username'],
+                    'title': document['topic_title'],
+                    'body': document['raw'],
+                    'body_type': 'raw',
+                    'created_at': document['updated_at'],
+                    'domain': "https://delvingbitcoin.org/",
+                    'url': f"https://delvingbitcoin.org/t/{document['topic_slug']}/{document['topic_id']}",
```

Comment:
because this is a thread we'll want to include the type field as well to know if it was an original post or a "reply"

Reply:
update script that will update the first post as "original" one, and "reply" to remaining ones.

Reply:
all of the fields should mimic the mailing list:

<img width="589" alt="Screenshot 2023-12-18 at 10 39 17 AM" src="https://github.com/bitcoinsearch/scraper/assets/755825/b578ad82-03bb-4157-88cf-e2be8e7fdd3a">

with the addition of body_formatted because it is written in markdown:
<img width="671" alt="Screenshot 2023-12-18 at 10 42 30 AM" src="https://github.com/bitcoinsearch/scraper/assets/755825/f317a831-bf4f-4099-9664-54c9faca07f0">